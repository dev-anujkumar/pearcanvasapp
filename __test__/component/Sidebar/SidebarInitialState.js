export default {
    "paragraph": {
        elementId: "urn:pearson:work:8a49e877-144a-4750-92d2-81d5188d8e1a",
        elementType: "element-authoredtext",
        elementWipType: "element-authoredtext",
        index: 1,
        primaryOption: "primary-paragraph",
        secondaryOption: "secondary-paragraph",
        tag: "P",
        toolbar: []
    },
    "assessment": {
        elementId: "urn:pearson:work:fa8ebea3-d9ba-4aec-8658-2f944841404a",
        elementType: "element-assessment",
        elementWipType: "figure",
        index: 3,
        primaryOption: "primary-single-assessment",
        secondaryOption: "secondary-single-assessment-cite",
        tag: "Qu",
        toolbar: ["bold", "italic", "underline", "strikethrough", "clearformatting", "increaseindent", "decreaseindent", "footnote", "glossary", "orderedlist", "unorderedlist", "mathml", "chemml", "inlinecode", "superscript", "subscript", "specialcharactor", "undo", "redo", "assetpopover", "slatetag"]
    },
    "image": {
        elementId: "urn:pearson:work:be34d78e-1638-4fcc-8511-5db7a3d4b395",
        elementType: "figure",
        elementWipType: "figure",
        index: 4,
        longDesc: "",
        primaryOption: "primary-image-figure",
        secondaryOption: "secondary-image-figure-width",
        tag: "Fg",
        toolbar: ["assetpopover", "decreaseindent", "glossary"]
    },
    "video": {
        elementId: "urn:pearson:work:e4d3e2ce-71f6-4ffa-9868-3d0b00a69f75",
        elementType: "video-audio",
        elementWipType: "figure",
        index: 5,
        primaryOption: "primary-video",
        secondaryOption: "secondary-video-alfresco",
        tag: "VID",
        toolbar: ["assetpopover", "glossary"]
    },
    "interactive": {
        altText: "",
        elementId: "urn:pearson:work:180226f6-ba40-406c-8b20-7842071ae4e3",
        elementType: "element-interactive",
        elementWipType: "figure",
        index: 6,
        primaryOption: "primary-mmi",
        secondaryOption: "secondary-interactive-mmi",
        tag: "MMI",
        toolbar: ["assetpopover", "glossary"]
    },
    "blockcode": {
        altText: "",
        elementId: "urn:pearson:work:e732173b-45a9-4859-a643-77d02abd8c7b",
        elementType: "figure",
        elementWipType: undefined,
        index: 7,
        longDesc: "",
        primaryOption: "primary-blockcode-equation",
        secondaryOption: "secondary-blockcode-language-Python",
        tag: "BCE",
        toolbar: ["assetpopover", "decreaseindent", "glossary"]
    },
    "list": {
        elementId: "urn:pearson:work:b922acc4-3ee7-4e77-bd6a-55b7bbe36d90",
        elementType: "element-authoredtext",
        elementWipType: "element-list",
        index: 8,
        primaryOption: "primary-list",
        secondaryOption: "secondary-list-2",
        tag: "OL",
        toolbar: []
    },
    "workedexample": {
        altText: "",
        elementId: "urn:pearson:manifest:f74c018e-16f1-4a3e-831c-d7e9c239610f",
        elementType: "element-workedexample",
        elementWipType: undefined,
        longDesc: "",
        primaryOption: "primary-workedexample-we1",
        secondaryOption: "secondary-workedexample-we1",
        tag: "WE",
        index: 9,
        toolbar: ["bold", "italic", "underline", "strikethrough", "clearformatting", "increaseindent", "decreaseindent", "footnote", "glossary", "orderedlist", "unorderedlist", "mathml", "chemml", "inlinecode", "superscript", "subscript", "specialcharactor", "undo", "redo", "assetpopover", "slatetag"]
    },
    "aside": {
        altText: "",
        longDesc: "",
        elementId: "urn:pearson:manifest:8e16f0f7-40d0-4abf-a515-4a3ac49122d5",
        elementType: "element-aside",
        elementWipType: "element-aside",
        index: 10,
        primaryOption: "primary-aside-aside",
        secondaryOption: "secondary-aside-sb1",
        tag: "As",
        toolbar: ["bold", "italic", "underline", "strikethrough", "clearformatting", "increaseindent", "decreaseindent", "footnote", "glossary", "orderedlist", "unorderedlist", "mathml", "chemml", "inlinecode", "superscript", "subscript", "specialcharactor", "undo", "redo", "assetpopover", "slatetag"]
    },
    "aside_authoredtext": {
        elementId: "urn:pearson:work:9ced8f63-da04-44a6-8ef1-19ecd1af77b1",
        elementType: "element-authoredtext",
        elementWipType: "element-authoredtext",
        index: "10-0",
        primaryOption: "primary-paragraph",
        secondaryOption: "secondary-paragraph",
        tag: "P",
        toolbar: []
    },
    "we_authoredtext": {
        elementId: "urn:pearson:work:fcba4843-df9e-491a-a144-b6a22fa5177a",
        elementType: "element-authoredtext",
        elementWipType: "element-authoredtext",
        index: "9-1-0",
        primaryOption: "primary-paragraph",
        secondaryOption: "secondary-paragraph",
        tag: "P",
        toolbar: []
    },
    "we_conversiondata": {
        tag: "H1",
        elementType: "element-authoredtext",
        primaryOption: "primary-heading",
        secondaryOption: "secondary-heading-1",
        elementId: "urn:pearson:work:b2852d7b-cc63-4085-bcf3-65c5d9bf2718",
        index: "0-0",
        elementWipType: "element-authoredtext",
        toolbar: ["bold", "underline", "strikethrough", "orderedlist", "unorderedlist", "increaseindent", "decreaseindent", "glossary", "assetpopover", "slatetag"]
    },
    "WE_OldData": {
        id: "urn:pearson:work:b2852d7b-cc63-4085-bcf3-65c5d9bf2718",
        type: "element-authoredtext",
        schema: "http://schemas.pearson.com/wip-authoring/element/1",
        designtype: "",
        elementdata:
        {
            schema: "http://schemas.pearson.com/wip-authoring/authoredtext/1#/definitions/authoredtext",
            text: "",
            headers: [{ charStart: 0, charEnd: -1, level: 1 }]
        },
        html:
        {
            text: '<h1 class="heading1NummerEins"><br></h1>',
            footnotes: {},
            glossaryentries: {}
        },
        versionUrn: "urn:pearson:work:b2852d7b-cc63-4085-bcf3-65c5d9bf2718",
        contentUrn: "urn:pearson:entity:0e2add65-fee2-4751-9346-fa1c924a1af3",
        status: "wip",
        tcm: false,
        feedback: false,
        comments: false
    },
    "WE_NewData": {
        elementId: "urn:pearson:work:b2852d7b-cc63-4085-bcf3-65c5d9bf2718",
        elementType: "element-authoredtext",
        primaryOption: "primary-paragraph",
        secondaryOption: "secondary-paragraph",
        labelText: "P",
        toolbar: []
    },
    "WE_OldInfo": {
        elementId: "urn:pearson:work:b2852d7b-cc63-4085-bcf3-65c5d9bf2718",
        index: "0-0",
        elementType: "element-authoredtext",
        primaryOption: "primary-heading",
        secondaryOption: "secondary-heading-1",
        tag: "H1",
        toolbar: ["bold", "underline", "strikethrough", "orderedlist", "unorderedlist", "increaseindent", "decreaseindent", "glossary", "assetpopover", "slatetag"],
        elementWipType: undefined,
        altText: "",
        longDesc: ""
    },
    "Section_ActiveElement": {
        elementId: "urn:pearson:work:b9125fbb-e982-4d5a-8bbf-748ad53f82d2",
        index: "0-1-0",
        elementType: "element-authoredtext",
        primaryOption: "primary-paragraph",
        secondaryOption: "secondary-paragraph",
        tag: "P"
    },
    "Section_OldData": {
        id: "urn:pearson:work:b9125fbb-e982-4d5a-8bbf-748ad53f82d2",
        type: "element-authoredtext",
        schema: "http://schemas.pearson.com/wip-authoring/element/1",
        designtype: "",
        elementdata:
        {
            schema: "http://schemas.pearson.com/wip-authoring/authoredtext/1#/definitions/authoredtext",
            text: ""
        },

        html:
        {
            text: '<p class="paragraphNumeroUno"><br></p>',
            footnotes: {},
            glossaryentries: {}
        },

        versionUrn: "urn:pearson:work:b9125fbb-e982-4d5a-8bbf-748ad53f82d2",
        contentUrn: "urn:pearson:entity:1c9395e3-6f50-452d-95c4-617c19f613dd",
        status: "wip",
        tcm: false,
        feedback: false,
        comments: false
    },
    "Section_NewData": {
        elementId: "urn:pearson:work:b9125fbb-e982-4d5a-8bbf-748ad53f82d2",
        elementType: "element-authoredtext",
        primaryOption: "primary-heading",
        secondaryOption: "secondary-heading-1",
        labelText: "H1",
        toolbar: ["bold", "underline", "strikethrough", "orderedlist", "unorderedlist", "increaseindent", "decreaseindent", "glossary", "assetpopover", "slatetag"]
    },
    "Section_OldInfo": {
        elementId: "urn:pearson:work:b9125fbb-e982-4d5a-8bbf-748ad53f82d2",
        index: "0-1-0",
        elementType: "element-authoredtext",
        primaryOption: "primary-paragraph",
        secondaryOption: "secondary-paragraph",
        tag: "P",
        toolbar: [],
        elementWipType: undefined,
        altText: "",
        longDesc: ""
    },
    "Aside_OldData": {
        id: "urn:pearson:work:188d143b-101c-400d-80f5-3102b25483bb",
        type: "element-authoredtext",
        schema: "http://schemas.pearson.com/wip-authoring/element/1",
        elementdata:
        {
            schema: "http://schemas.pearson.com/wip-authoring/authoredtext/1#/definitions/authoredtext",
            text: ""
        },
        html:
        {
            text: '<p class="paragraphNumeroUno"><br></p>'
        },
        versionUrn: "urn:pearson:work:188d143b-101c-400d-80f5-3102b25483bb",
        contentUrn: "urn:pearson:entity:374dc898-adc7-4dd7-a4f9-224fafb0215c",
        status: "wip",
        tcm: false,
        feedback: false,
        comments: false
    },
    "Aside_NewData": {
        elementId: "urn:pearson:work:188d143b-101c-400d-80f5-3102b25483bb",
        elementType: "element-authoredtext",
        primaryOption: "primary-heading",
        secondaryOption: "secondary-heading-1",
        labelText: "H1",
        toolbar: ["bold", "underline", "strikethrough", "orderedlist", "unorderedlist", "increaseindent", "decreaseindent", "glossary", "assetpopover", "slatetag"]
    },
    "Aside_OldInfo": {
        tag: "P",
        elementType: "element-authoredtext",
        primaryOption: "primary-paragraph",
        secondaryOption: "secondary-paragraph",
        elementId: "urn:pearson:work:188d143b-101c-400d-80f5-3102b25483bb",
        index: "0-1",
        elementWipType: "element-authoredtext",
        toolbar: []
    },
    "Aside_ActiveElement": {
        tag: "P",
        elementType: "element-authoredtext",
        primaryOption: "primary-paragraph",
        secondaryOption: "secondary-paragraph",
        elementId: "urn:pearson:work:188d143b-101c-400d-80f5-3102b25483bb",
        index: "0-1",
        elementWipType: "element-authoredtext"
    },
    "Aside_MMI_ActiveElement": {
        elementId: "urn:pearson:work:c1fc3bef-e90c-46ac-8f99-fad2df981611",
        index: "0-2",
        elementType: "element-interactive",
        primaryOption: "primary-mmi",
        secondaryOption: "secondary-interactive-mmi",
        tag: "MMI",
        toolbar: ["assetpopover", "glossary"],
        elementWipType: undefined,
        altText: "",
        longDesc: ""
    },
    "Aside_MMI_OldData": {
        id: "urn:pearson:work:c1fc3bef-e90c-46ac-8f99-fad2df981611",
        type: "figure",
        schema: "http://schemas.pearson.com/wip-authoring/figure/1",
        versionUrn: "urn:pearson:work:c1fc3bef-e90c-46ac-8f99-fad2df981611",
        contentUrn: "urn:pearson:entity:6f2e78b9-afe5-40ed-9017-b18dbd8e82fb",
        title: {
            schema: "http://schemas.pearson.com/wip-authoring/authoredtext/1#/definitions/authoredtext",
            text: "",
            textsemantics: [],
            mathml: []
        },
        captions: {
            schema: "http://schemas.pearson.com/wip-authoring/authoredtext/1#/definitions/authoredtext",
            text: "",
            textsemantics: [],
            mathml: []
        },
        credits: {
            schema: "http://schemas.pearson.com/wip-authoring/authoredtext/1#/definitions/authoredtext",
            text: "",
            textsemantics: [],
            mathml: []
        },
        feedback: false,
        comments: false,
        status: "wip",
        figuretype: "interactive",
        subtitle: {
            schema: "http://schemas.pearson.com/wip-authoring/authoredtext/1#/definitions/authoredtext",
            text: "",
            textsemantics: [],
            mathml: []
        },
        figuredata: {
            schema: "http://schemas.pearson.com/wip-authoring/interactive/1#/definitions/interactive",
            interactiveid: "",
            interactivetype: "showhide",
            interactiveformat: "mmi"
        },
        html: {
            title: "<p class='paragraphNumeroUno'><br></p>",
            subtitle: "<p class='paragraphNumeroUno'><br></p>",
            text: "",
            postertext: "",
            captions: "<p class='paragraphNumeroUno'><br></p>",
            credits: "<p class='paragraphNumeroUno'><br></p>",
        }
    },
    "Aside_MMI_NewData": {
        elementId: "urn:pearson:work:c1fc3bef-e90c-46ac-8f99-fad2df981611",
        elementType: "element-interactive",
        primaryOption: "primary-showhide",
        secondaryOption: "secondary-aside-showhide",
        labelText: "SH",
        toolbar: ["footnote", "glossary", "assetpopover"]
    },
    "Aside_MMI_OldInfo": {
        elementId: "urn:pearson:work:c1fc3bef-e90c-46ac-8f99-fad2df981611",
        index: "0-2",
        elementType: "element-interactive",
        primaryOption: "primary-mmi",
        secondaryOption: "secondary-interactive-mmi",
        tag: "MMI",
        toolbar: ["assetpopover", "glossary"],
        elementWipType: undefined,
        altText: "",
        longDesc: ""
    }
}