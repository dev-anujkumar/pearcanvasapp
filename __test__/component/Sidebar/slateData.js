export default {
    "urn:pearson:manifest:d9023151-3417-4482-8175-fc965466220e": {
        "id": "urn:pearson:manifest:d9023151-3417-4482-8175-fc965466220e",
        "schema": "http://schemas.pearson.com/wip-authoring/manifest/1",
        "type": "manifest",
        "contents": {
            "schema": "http://schemas.pearson.com/wip-authoring/manifest/1#/definitions/manifest",
            "title": {
                "schema": "http://schemas.pearson.com/wip-authoring/authoredtext/1#/definitions/authoredtext",
                "text": "Test Slate 2"
            },
            "frontmatter": [],
            "bodymatter": [
                {
                    "id": "urn:pearson:work:8a49e877-144a-4750-92d2-81d5188d8e1b",
                    "type": "element-authoredtext",
                    "subtype": "",
                    "schema": "http://schemas.pearson.com/wip-authoring/element/1",
                    "elementdata": {
                        "schema": "http://schemas.pearson.com/wip-authoring/authoredtext/1#/definitions/authoredtext",
                        "headers": [
                            {
                                "charStart": 0,
                                "charEnd": 8,
                                "level": 1
                            }
                        ],
                        "text": "Heading 1"
                    },
                    "html": {
                        "text": "<h1 class=\"heading1NummerEins\">Heading 1</h1>"
                    },
                    "comments": false,
                    "tcm": false,
                    "versionUrn": "urn:pearson:work:8a49e877-144a-4750-92d2-81d5188d8e0a",
                    "contentUrn": "urn:pearson:entity:b70a5dbe-cc3b-456d-87fc-e369ac59c527"
                },
                {
                    "id": "urn:pearson:work:8a49e877-144a-4750-92d2-81d5188d8e1a",
                    "type": "element-authoredtext",
                    "subtype": "",
                    "schema": "http://schemas.pearson.com/wip-authoring/element/1",
                    "elementdata": {
                        "schema": "http://schemas.pearson.com/wip-authoring/authoredtext/1#/definitions/authoredtext",
                        "text": "Paragraph"
                    },
                    "html": {
                        "text": "<p class=\"paragraphNumeroUno\">Paragraph</p>"
                    },
                    "comments": false,
                    "tcm": false,
                    "feedback": true,
                    "versionUrn": "urn:pearson:work:8a49e877-144a-4750-92d2-81d5188d8e0a",
                    "contentUrn": "urn:pearson:entity:b70a5dbe-cc3b-456d-87fc-e369ac59c527"
                },
                {
                    "id": "urn:pearson:work:8a49e877-144a-4750-92d2-81d5188d8e1c",
                    "type": "element-authoredtext",
                    "subtype": "",
                    "schema": "http://schemas.pearson.com/wip-authoring/element/1",
                    "elementdata": {
                        "schema": "http://schemas.pearson.com/wip-authoring/authoredtext/1#/definitions/authoredtext",
                        "headers": [
                            {
                                "charStart": 0,
                                "charEnd": 8,
                                "level": 4
                            }
                        ],
                        "text": "Heading 4"
                    },
                    "html": {
                        "text": "<h4 class=\"heading4NummerEins\">Heading 4</h4>"
                    },
                    "comments": false,
                    "tcm": false,
                    "versionUrn": "urn:pearson:work:8a49e877-144a-4750-92d2-81d5188d8e0a",
                    "contentUrn": "urn:pearson:entity:b70a5dbe-cc3b-456d-87fc-e369ac59c527"
                }
            ]
        }
    }
}