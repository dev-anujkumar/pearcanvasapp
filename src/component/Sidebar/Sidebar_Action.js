import axios from 'axios';
import config  from './../../config/config';
import {
    FETCH_SLATE_DATA,
    SET_ACTIVE_ELEMENT
} from './../../constants/Action_Constants';
import elementTypes from './../Sidebar/elementTypes';
import figureDataBank from '../../js/figure_data_bank';
import { sendDataToIframe } from '../../constants/utility.js';
let imageSource = ['image','table','mathImage'],imageDestination = ['primary-image-figure','primary-image-table','primary-image-equation']

const convertElement = (oldElementData, newElementData, oldElementInfo, store, indexes) => dispatch => {
    try {
        sendDataToIframe({ 'type': 'isDirtyDoc', 'message': { isDirtyDoc: true } })
    // Input Element
    const inputPrimaryOptionsList = elementTypes[oldElementInfo['elementType']],
        inputPrimaryOptionType = inputPrimaryOptionsList[oldElementInfo['primaryOption']],
        overallType = inputPrimaryOptionsList['enumType']

    const inputSubTypeList = inputPrimaryOptionType['subtype'],
        inputSubType = inputSubTypeList[[oldElementInfo['secondaryOption']]]

    let inputSubTypeEnum = inputSubType['enum'],
    inputPrimaryOptionEnum = inputPrimaryOptionType['enum']

    if(oldElementData.figuretype==="assessment"){
        inputPrimaryOptionEnum=inputSubType['enum'];
        inputSubTypeEnum=document.querySelector(`div[data-id='${oldElementData.id}'] span.singleAssessment_Dropdown_currentLabel`).innerText.toUpperCase().replace(" ", "_").replace("-", "_");
    }
    
    // Output Element
    const outputPrimaryOptionsList = elementTypes[newElementData['elementType']],
        outputPrimaryOptionType = outputPrimaryOptionsList[newElementData['primaryOption']]

    const outputSubTypeList = outputPrimaryOptionType['subtype'],
        outputSubType = outputSubTypeList[[newElementData['secondaryOption']]]

    if (oldElementData.type === "figure") {
        if (!(imageSource.includes(oldElementData.figuretype) && imageDestination.includes(newElementData['primaryOption']))){
            oldElementData.figuredata = {...figureDataBank[newElementData['primaryOption']]}
        }
        if(oldElementData.figuredata.srctype){
            oldElementData.figuredata.srctype=outputSubType['wipValue']
        }
        if(oldElementData.figuredata.interactivetype){
            oldElementData.figuredata.interactivetype=outputSubType['wipValue'];
        }

        if(oldElementData.figuretype && oldElementData.figuretype === "codelisting" && newElementData['primaryOption'] === "primary-blockcode-equation") {
            oldElementData.figuredata.programlanguage = elementTypes[newElementData['elementType']][newElementData['primaryOption']].subtype[newElementData['secondaryOption']].text;
        }
    }

    let outputSubTypeEnum = outputSubType['enum'],
    outputPrimaryOptionEnum = outputPrimaryOptionType['enum']
    if (oldElementData.figuretype === "assessment") {
        let usageType=document.querySelector(`div[data-id='${oldElementData.id}'] span.singleAssessment_Dropdown_currentLabel`).innerText;
        outputPrimaryOptionEnum=outputSubType['enum'],
        outputSubTypeEnum = usageType.toUpperCase().replace(" ", "_").replace("-", "_"),
        oldElementData.figuredata.elementdata.usagetype=usageType;
        let assessmentFormat =outputSubType.text.toLowerCase();
        let assessmentItemType ="";
        if(assessmentFormat==="cite"){
            assessmentItemType ="assessmentItem";
        }else{
            assessmentItemType = "tdxAssessmentItem";
        }        
        oldElementData.html.title = "";
        oldElementData.figuredata.elementdata.assessmentformat=assessmentFormat
        oldElementData.figuredata.elementdata.assessmentitemtype=assessmentItemType;
    }
    /**
     * Patch [code in If block] - in case list is being converted from toolbar and there are some unsaved changes in current element
     * then send dom html data instead of sending store data
     */
    if (newElementData.primaryOption === "primary-list") {
        let storeHtml = oldElementData.html && oldElementData.html.text || ''
        let containerDom = document.querySelector(`[data-id='${oldElementData.id}']`)
        let elementContainer = containerDom && containerDom.querySelector('.element-container')
        let editableDom = elementContainer && elementContainer.querySelector('.cypress-editable')
        let domHtml = editableDom.innerHTML
        if (storeHtml !== domHtml) {
            oldElementData.html.text = domHtml
        }
        /**
         * case - if bullet list is being converted into bullet again then explicitly proceed with paragraph coversion
         */
        if (inputPrimaryOptionEnum === outputPrimaryOptionEnum && inputSubTypeEnum === outputSubTypeEnum && outputSubTypeEnum === "DISC") {
            outputPrimaryOptionEnum = "AUTHORED_TEXT"
            outputSubTypeEnum = "NA"
        }
    }
    const conversionDataToSend = {
        ...oldElementData,
        inputType : inputPrimaryOptionEnum,
        inputSubType : inputSubTypeEnum,
        outputType : outputPrimaryOptionEnum,
        outputSubType: outputSubTypeEnum,
        projectUrn : config.projectUrn,
        slateUrn:config.slateManifestURN,
        counterIncrement: (newElementData.startvalue > 0) ? (newElementData.startvalue - 1) : 0
    }

    
    const url = `${config.REACT_APP_API_URL}v1/slate/elementTypeConversion/${overallType}`
    axios.post(url, JSON.stringify(conversionDataToSend), { 
        headers: {
			"Content-Type": "application/json",
			"PearsonSSOSession": config.ssoToken
		}
    }).then(res =>{
        sendDataToIframe({ 'type': 'isDirtyDoc', 'message': { isDirtyDoc: false } })
        let storeElement = store[config.slateManifestURN];
        let bodymatter = storeElement.contents.bodymatter;
        let focusedElement = bodymatter;
        indexes.forEach(index => {
            if(newElementData.elementId === focusedElement[index].id) {
                focusedElement[index] = res.data;
            } else {
                if(('elementdata' in focusedElement[index] && 'bodymatter' in focusedElement[index].elementdata) || ('contents' in focusedElement[index] && 'bodymatter' in focusedElement[index].contents)) {
                  //  focusedElement = focusedElement[index].elementdata.bodymatter;
                    focusedElement = focusedElement[index].elementdata && focusedElement[index].elementdata.bodymatter ||  focusedElement[index].contents.bodymatter
                }
            }
        });
        store[config.slateManifestURN].contents.bodymatter = bodymatter;//res.data;
        let altText="";
        let longDesc="";
        if(res.data.figuredata){
            altText=res.data.figuredata && res.data.figuredata.alttext ? res.data.figuredata.alttext : "";
            longDesc = res.data.figuredata && res.data.figuredata.longdescription ? res.data.figuredata.longdescription : "";
        }
        let activeElementObject = {
            elementId: newElementData.elementId,
            index: indexes.join("-"),
            elementType: newElementData.elementType,
            primaryOption: newElementData.primaryOption,
            secondaryOption: newElementData.secondaryOption,
            tag: newElementData.labelText,
            toolbar: newElementData.toolbar,
            elementWipType: newElementData.elementWipType,
            altText,
            longDesc
        };

        dispatch({
            type: FETCH_SLATE_DATA,
            payload: store
        });

        dispatch({
            type: SET_ACTIVE_ELEMENT,
            payload: activeElementObject
        });
    })
    .catch(err =>{
        sendDataToIframe({ 'type': 'isDirtyDoc', 'message': { isDirtyDoc: false } })
        //console.log(err) 
    })
}
catch (error) {
    sendDataToIframe({ 'type': 'isDirtyDoc', 'message': { isDirtyDoc: false } })
}
}

const handleElementConversion = (elementData, store, activeElement) => dispatch => {
    store = JSON.parse(JSON.stringify(store));
    if(Object.keys(store).length > 0 && config.slateManifestURN === Object.keys(store)[0]) {
        let storeElement = store[config.slateManifestURN];
        let bodymatter = storeElement.contents.bodymatter;
        let indexes = activeElement.index;
        indexes = indexes.toString().split("-");
        
        indexes.forEach(index => {
            if(elementData.elementId === bodymatter[index].id) {
                dispatch(convertElement(bodymatter[index], elementData, activeElement, store, indexes));
            } else {
                if(('elementdata' in bodymatter[index] && 'bodymatter' in bodymatter[index].elementdata) || ('contents' in bodymatter[index] && 'bodymatter' in bodymatter[index].contents))  {
                    bodymatter = bodymatter[index].elementdata && bodymatter[index].elementdata.bodymatter ||  bodymatter[index].contents.bodymatter
                }
                
            }
        });
    }
    
    return store;
}

export const conversionElement = (elementData) => (dispatch, getState) => {
    let appStore =  getState().appStore;
    dispatch(handleElementConversion(elementData, appStore.slateLevelData, appStore.activeElement));
}