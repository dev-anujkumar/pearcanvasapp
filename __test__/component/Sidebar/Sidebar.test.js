import React from 'react';
import { mount } from 'enzyme';
import Sidebar from '../../../src/component/Sidebar';
import { updateElement } from './../../../src/component/Sidebar/Sidebar_Action';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
const middlewares = [thunk];
import { Provider } from 'react-redux';
import slateLevelData from './slateData';

describe('Test for Sidebar component', () => {
    const mockStore = configureMockStore(middlewares);

    const activeElement = {
        elementId: "urn:pearson:work:8a49e877-144a-4750-92d2-81d5188d8e1b",
        elementType: "element-authoredtext",
        elementWipType: "element-authoredtext",
        primaryOption: "primary-heading",
        secondaryOption: "secondary-heading-1",
        index: "1-0",
        tag: "H1"
    };
    
    const sidebarWithData = mockStore({
        appStore: {
            activeElement,
            updateElement,
            slateLevelData,
        },
        metadataReducer: {
            showModule: true
        },
    });
    let props = {
        slateId: 'urn:pearson:manifest:e652706d-b04b-4111-a083-557ae121af0f'
    };

    let sidebar = mount(<Provider store={sidebarWithData}>
        <Sidebar />
    </Provider>);

    it('onClick Event', () => {
        const sidebarInstance = sidebar.find('Sidebar').instance();
        sidebarInstance.setState({
            elementDropdown: 'primary',
            activeElementType: 'element-authoredtext',
            activePrimaryOption: 'primary-paragraph',
            activeSecondaryOption: 'secondary-paragraph',
            activeLabelText: 'P'
        });

        let target = {
            target: {
                getAttribute: function(dataValue) {
                    return 'primary-heading';
                }
            }
        }

        sidebar.find('div.element-dropdown-title').at(0).simulate('click');
        sidebar.find('ul.element-dropdown-content.primary-options').simulate('click');
        sidebarInstance.handlePrimaryOptionChange(target);

        sidebarInstance.setState({
            elementDropdown: 'secondary'
        });

        target = {
            target: {
                getAttribute: function(dataValue) {
                    return 'secondary-heading-1';
                }
            }
        }

        sidebar.find('ul.element-dropdown-content.secondary-options').simulate('click');
        sidebarInstance.handleSecondaryOptionChange(target);

        // Attribution for secondary element type
        sidebarInstance.setState({
            elementDropdown: 'primary',
            activePrimaryOption: 'primary-blockquote',
            activeSecondaryOption: 'secondary-marginalia-attribution'
        });

        sidebarInstance.attributions();

        // Attribution for primary element type
        sidebarInstance.setState({
            activeElementType: 'figure',
            activePrimaryOption: 'primary-image-figure',
            activeSecondaryOption: 'secondary-image-figure-half'
        });

        sidebarInstance.attributions();
    });

    it('Test Case for Metadata Anchor LO', () => {
        const activeElement = {
            elementId: "urn:pearson:work:8a49e877-144a-4750-92d2-81d5188d8e1b",
            elementType: "",
            elementWipType: "element-learningobjectivemapping",
            primaryOption: "",
            secondaryOption: "",
            index: 2,
            tag: "LO"
        };

        const sidebarWithData = mockStore({
            appStore: {
                activeElement,
                updateElement,
                slateLevelData
            },
            metadataReducer: {
                currentSlateLOData: {}
            },
        });

        let sidebar = mount(<Provider store={sidebarWithData}>
            <Sidebar />
        </Provider>);
    });

    it('Test Case for Metadata Anchor MA', () => {
        const activeElement = {
            elementId: "urn:pearson:work:8a49e877-144a-4750-92d2-81d5188d8e1b",
            elementType: "",
            elementWipType: "element-generateLOlist",
            primaryOption: "",
            secondaryOption: "",
            index: 1,
            tag: "LO"
        };

        const sidebarWithData = mockStore({
            appStore: {
                activeElement,
                updateElement,
                slateLevelData
            },
            metadataReducer: {
                currentSlateLOData: {}
            },
        });

        let sidebar = mount(<Provider store={sidebarWithData}>
            <Sidebar />
        </Provider>);
    });

    it("With no activeElement", () => {
        const activeElement = {}
        
        const sidebarWithData = mockStore({
            appStore: {
                activeElement,
                updateElement,
                slateLevelData
            },
            metadataReducer: {
                currentSlateLOData: {}
            },
        });
    let sidebar = mount(<Provider store={sidebarWithData}>
            <Sidebar />
        </Provider>);
    })

    describe("Blockquote", () => {

        it("Checking data for pullquote", () => {
            const activeElement = {
                elementId: "urn:pearson:work:28493c52-4356-48e5-8328-c24337bb3200",
                elementType: "element-authoredtext",
                elementWipType: "element-blockfeature",
                index: 0,
                primaryOption: "primary-blockquote",
                secondaryOption: "secondary-pullquote",
                tag: "BQ"
            }
            
            const sidebarWithData = mockStore({
                appStore: {
                    activeElement,
                    updateElement,
                    slateLevelData
                },
                metadataReducer: {
                    currentSlateLOData: {}
                },
            });
        let sidebar = mount(<Provider store={sidebarWithData}><Sidebar /></Provider>);
            expect(sidebar.find('.element-dropdown').length).toBe(2)
            expect(sidebar.find('.element-dropdown-title[data-element="primary"]').text()).toBe("Blockquotes")
            expect(sidebar.find('.element-dropdown-title[data-element="secondary"]').text()).toBe("Pullquote")
        })

        it("Checking data for Marginalia with attribution", () => {
            const activeElement = {
                elementId: "urn:pearson:work:8a49e877-144a-4750-92d2-81d5188d8e2t",
                elementType: "element-authoredtext",
                elementWipType: "element-blockfeature",
                primaryOption: "primary-blockquote",
                secondaryOption: "secondary-marginalia-attribution",
                index: 1,
                tag: "BQ"
            }
            
            const sidebarWithData = mockStore({
                appStore: {
                    activeElement,
                    updateElement,
                    slateLevelData
                },
                metadataReducer: {
                    currentSlateLOData: {}
                },
            });
            let sidebar = mount(<Provider store={sidebarWithData}><Sidebar /></Provider>);
            expect(sidebar.find('.element-dropdown').length).toBe(2)
            expect(sidebar.find('.element-dropdown-title[data-element="primary"]').text()).toBe("Blockquotes")
            expect(sidebar.find('.element-dropdown-title[data-element="secondary"]').text()).toBe("Marginalia with Attribution")
        }) 
    })
});