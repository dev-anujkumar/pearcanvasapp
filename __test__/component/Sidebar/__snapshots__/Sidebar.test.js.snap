// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Sidebar component Render element container  1`] = `
<Sidebar
  elementType="element-authoredtext"
>
  <div
    className="canvas-sidebar"
  >
    <div
      className="canvas-sidebar-heading"
    >
      Settings
    </div>
    <div
      className="element-dropdown"
    >
      <div
        className="element-dropdown-title"
        data-element="primary"
        onClick={[Function]}
      >
        Paragraph
        <svg
          className="dropdown-arrow"
          viewBox="0 0 9 4.5"
        >
          <path
            d="M0,0,4.5,4.5,9,0Z"
          />
        </svg>
      </div>
      <ul
        className="element-dropdown-content primary-options "
      >
        <li
          data-value="primary-paragraph"
          key="primary-paragraph"
          onClick={[Function]}
        >
          Paragraph
        </li>
        <li
          data-value="primary-heading"
          key="primary-heading"
          onClick={[Function]}
        >
          Headings
        </li>
        <li
          data-value="primary-learning-objective"
          key="primary-learning-objective"
          onClick={[Function]}
        >
          Learning Objective Item
        </li>
        <li
          data-value="primary-blockquote"
          key="primary-blockquote"
          onClick={[Function]}
        >
          Blockquotes
        </li>
        <li
          data-value="primary-list"
          key="primary-list"
          onClick={[Function]}
        >
          List
        </li>
      </ul>
    </div>
  </div>
</Sidebar>
`;
